### DISCUSSION TITLE
[Question] Unique IDs in yaml?

### DISCUSSION THREAD

**Discussion by ddppddpp** [frenck/home-assistant-config#63] [Q&A]
Hi frenck,
How do you use your split-folder/one-file-per-thing config system with unique IDs and do they allow you to manipulate i.e. an automation in GUI?
Thanks!

**Comment by frenck** [comment]
The Unique ID's allow for some additional control in Home Assistant. For example, it allows for overriding names, icon,s and assigning areas from the UI.

It does not provide capabilities for manipulating automations from the UI.

**Reply by ddppddpp** [reply]
Got it, thanks, close enough to 'working with the GUI' for me :-)
Can I use like any web based UUID tool to generate them?

**Reply by frenck** [reply]
they can be anything, as long as it is unique to your system. E.g., you can number them, name them or a UUID as I do.

<https://www.uuidgenerator.net/version4>

There are also VSCode extensions available that can do that (if VSCode is your goto editor)
